src/model/statement.js

```diff
@@ -0,0 +1,356 @@
+    if (Buffer.isBuffer(stmt.signature)) {
+      this.signature = (stmt.signature: any)
+    } else {
+      this.signature = Buffer.from(stmt.signature, 'base64')
+    }
+  }
+
+  static create (
+    publisherId: PublisherId,
+    namespace: string,
+    statementBody: StatementBody | StatementBodyMsg,
+    counter: number = 0,
+    timestampGenerator: () => number = Date.now)
+  : Promise<Statement> {
+    const body = (statementBody instanceof StatementBody) ? statementBody : StatementBody.fromProtobuf(statementBody)
+    const timestamp = timestampGenerator()
+    const statementId = [publisherId.id58, timestamp.toString(), counter.toString()].join(':')
+    const stmt = new Statement({
+      id: statementId,
+      publisher: publisherId.id58,
+      namespace,
+      timestamp,
+      body,
+      signature: Buffer.from('')
+    })
+    return stmt.sign(publisherId)
+  }
+
+  static createSimple (
+    publisherId: PublisherId,
+    namespace: string,
+    statementBody: {object: string | Object, refs?: Array<string>, deps?: Array<string>, tags?: Array<string>},
+    counter: number = 0,
+    timestampGenerator: () => number = Date.now
+  ): Promise<Statement> {
+    let body
+    if (typeof statementBody.object === 'object') {
+      body = new ExpandedSimpleStatementBody((statementBody: any))
+    } else {
+      body = new SimpleStatementBody((statementBody: any))
+    }
+    return Statement.create(publisherId, namespace, body, counter, timestampGenerator)
+  }
+
+  static fromProtobuf (msg: StatementMsg): Statement {
+    const body = StatementBody.fromProtobuf(msg.body)
+
+    const {id, publisher, namespace, timestamp, signature} = msg
+    return new Statement({id, publisher, namespace, timestamp, body, signature})
+  }
+
+  static fromBytes (msgBuffer: Buffer): Statement {
+    return Statement.fromProtobuf(pb.stmt.Statement.decode(msgBuffer))
+  }
+
+  toProtobuf (): StatementMsg {
+    const {id, publisher, namespace, timestamp, signature, body} = this
+    return {id, publisher, namespace, timestamp: timestamp, signature, body: body.toProtobuf()}
+  }
+
+  toBytes (): Buffer {
+    return pb.stmt.Statement.encode(this.toProtobuf())
+  }
+
+  toJSON (): Object {
+    return stringifyNestedBuffers(this.toProtobuf())
+  }
+
+  inspect (_depth?: number, opts?: Object) {
+    opts = Object.assign({}, opts, {depth: null})
+    const {id, publisher, namespace, timestamp, signature, body} = this
+    const output = stringifyNestedBuffers({id, publisher, namespace, timestamp, signature, body})
+    return inspect(output, opts)
+  }
+
+  get objectIds (): Array<string> {
+    return this.body.objectIds
+  }
+
+  get refSet (): Set<string> {
+    return this.body.refSet
+  }
+
+  get source (): string {
+    if (!(this.body instanceof EnvelopeStatementBody)) {
+      return this.publisher
+    }
+
+    if (this.body.statements.length < 1) {
+      return this.publisher
+    }
+
+    return this.body.statements[0].source
```

Comment:
I was thinking about the semantics of this, don't think we have a good way to frame it so I guess this is fine for now